test_expressions/test_expressions.cc

```diff
@@ -0,0 +1,129 @@
+#include <gtest/gtest.h>
+
+#include "../ky_expressions/expressions.h"
+#include "../ky_expressions/expressions_dd.h"
+
+// TODO(ashish): see if you can easily populate the folloing driver
+//               this will give us common interface we can target.
+//
+//               In theory we can define a concept that both our
+//               implementations satisfy, but it is an overkill for now.
+//
+//               You can see my implementation of the driver right under.
+//               If you want to change something reasonable about the
+//               driver interface, let me know and I will adjust mine.
+//
+#ifdef ASHISH_POPULATED_DRIVER
+
+// Ashish's Driver (no dynamic dispatch)
+class AtExpressionsDriver {
+  using Expr = ...
+
+      public : int
+               compute(Expr e) { /* ... */
+  }
+  void Save(Expr e, std::ostream &s) { /* ... */
+  }
+  Expr Load(std::istream &s) { /* ... */
+  }
+  Expr ConstantExpression(int value) { /* ... */
+  }
+  Expr BinaryOperatorExpression(char op, Expr l, Expr r) { /* ... */
+  }
+}
+#endif
```

Comment:
this is the part that you need to fill out so that we can run the same tests without modification.